cpp/src/neighbors/detail/cagra/search_single_cta_kernel-inl.cuh

```diff
@@ -863,15 +864,37 @@ __device__ void search_core(
     __syncthreads();
   }
 
+  // NB: tag the indices pointer with its element size.
+  //     This allows us to avoid multiplying kernel instantiations
+  //     and any costs for extra registers in the kernel signature.
```

Comment:
```suggestion
  // NB: The indices pointer is tagged with its element size.
  //     Here we select the correct conversion operator at runtime.
  //     This allows us to avoid multiplying kernel instantiations
  //     and any costs for extra registers in the kernel signature.
```